src/colorizer/ColorizeCanvas3D.ts

```diff
@@ -430,4 +448,58 @@ export class ColorizeCanvas3D implements IRenderCanvas {
     }
     return null;
   }
+
+  public getScreenSpaceMatrix(): Matrix4 {
+    if (!this.volume) {
+      // Return an identity matrix if the volume is not loaded
+      return new Matrix4();
+    }
+
+    // 1. Normalize from volume voxel coordinates to world space. Also,
+    //    translate so that the center of the volume is at (0, 0, 0).
+    const volumeScale = new Vector3(1, 1, 1)
+      .multiply(this.volume.physicalPixelSize)
+      .divideScalar(this.volume.physicalScale);
+    const normalizeVoxelToWorld = new Matrix4().compose(
+      this.volume.normPhysicalSize.clone().multiplyScalar(-0.5), // Translate to center
+      new Quaternion(0, 0, 0, 1),
+      volumeScale
+    );
+
+    // 2. Get the view projection matrix, which transforms from world space to
+    //    screen space in the [-1, 1] range.
+    const viewProjectionMatrix = this.view3d.getViewProjectionMatrix();
+
+    // 3. Scale the [-1, 1] range to canvas pixels, and move the origin to the
+    //    top left corner of the canvas. Normalize the Z axis to the [0, 1] range.
+    const viewProjectionToScreen = new Matrix4().compose(
+      new Vector3(0.5 * this.canvasResolution.x, 0.5 * this.canvasResolution.y, 0.5), // Translate origin
+      new Quaternion(0, 0, 0, 1),
+      new Vector3(0.5 * this.canvasResolution.x, -0.5 * this.canvasResolution.y, 0.5) // Scale to screen
+    );
+
+    return viewProjectionToScreen.multiply(viewProjectionMatrix).multiply(normalizeVoxelToWorld);
+  }
+
+  public getDepthToScaleFn(screenSpaceMatrix: Matrix4): (depth: number) => { scale: number; clipOpacity: number } {
+    if (!this.volume) {
+      return () => ({ scale: 1, clipOpacity: 1 });
+    }
+    // Determine min and max Z depth of the volume in screen space, using the
+    // corners.
+    const box3d = new Box3(new Vector3(0, 0, 0), this.volume.imageInfo.originalSize.clone().subScalar(1));
+    box3d.applyMatrix4(screenSpaceMatrix);
+    const minZ = Math.max(0.01, Math.min(box3d.min.z, box3d.max.z));
+    const maxZ = Math.min(1, Math.max(box3d.min.z, box3d.max.z));
```

Comment:
AFAIK the box should already have its min and max extents determined... You can check to see if this is true.
```suggestion
    const maxZ = Math.min(1, box3d.max.z);
```